pkg/metrics/stats_reporter.go

```diff
@@ -0,0 +1,101 @@
+package metrics
+
+import (
+	"context"
+	"runtime"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/global"
+)
+
+var (
+	instrumentationName = "keyvaultkms"
+	osTypeKey           = "os_type"
```

Comment:
The os type key might not be relevant here. The csi driver could run on linux/windows but the kms plugin is only for Linux as it runs in the master.

Reply:
removed